mountpoint-s3/src/cli.rs

```diff
@@ -877,6 +877,12 @@ where
 
     let (client, runtime, s3_personality) = client_builder(&args)?;
 
+    if args.incremental_upload && !matches!(s3_personality, S3Personality::ExpressOneZone) {
+        return Err(anyhow!(
+            "--incremental-upload is only supported for S3 Express One Zone buckets"
+        ));
+    }
```

Comment:
This should be below the log on line 887 that describes the detected bucket type